types/index.d.ts

```diff
@@ -0,0 +1,224 @@
+    'resultMatch' |
+    'resultNoMatch' |
+    'errorNetwork' |
+    'errorPermissionBlocked' |
+    'errorPermissionDenied';
+
+export interface Annyang {
+    /**
+     * Start listening.
+     * It's a good idea to call this after adding some commands first, but not mandatory.
+     *
+     * @param {StartOptions} options
+     */
+    start(options?: StartOptions): void;
+
+    /**
+     * Stop listening, and turn off mic.
+     *
+     */
+    abort(): void;
+
+    /**
+     * Pause listening. annyang will stop responding to commands (until the resume or start methods are called), without turning off the browser's SpeechRecognition engine or the mic.
+     *
+     */
+    pause(): void;
+
+    /**
+     * Resumes listening and restores command callback execution when a result matches.
+     * If SpeechRecognition was aborted (stopped), start it.
+     *
+     */
+    resume(): void;
+
+    /**
+     * Turn on output of debug messages to the console. Ugly, but super-handy!
+     *
+     * @export
+     * @param {boolean} [newState=true] Turn on/off debug messages
+     */
+    debug(newState?: boolean): void;
+
+    /**
+     * Set the language the user will speak in. If this method is not called, defaults to 'en-US'.
+     *
+     * @param {string} lang
+     * @see [Languages](https://github.com/TalAter/annyang/blob/master/docs/FAQ.md#what-languages-are-supported)
+     */
+    setLanguage(lang: string): void;
+
+    /**
+     * Add commands that annyang will respond to. Similar in syntax to init(), but doesn't remove existing commands.
+     *
+     * #### Examples:
+     * ````javascript
+     * var commands = {'hello :name': helloFunction, 'howdy': helloFunction};
+     * var commands2 = {'hi': helloFunction};
+     *
+     * annyang.addCommands(commands);
+     * annyang.addCommands(commands2);
+     * // annyang will now listen to all three commands
+     * ````
+     *
+     * @param {CommandOption} commands
+     */
+    addCommands(commands: CommandOption): void;
+
+    /**
+     * Removes all existing commands or a specific command
+     * #### Examples:
+     * ````javascript
+     * var commands : annyang.CommandOption = {'hello': helloFunction, 'howdy': helloFunction, 'hi': helloFunction};
+     *
+     * // Don't respond to hello
+     * annyang.removeCommands('hello');
+     *
+     * // Remove all existing commands
+     * annyang.removeCommands();
+     * ````
+     * @param {string} command
+     */
+    removeCommands(command?: string): void;
+
+    /**
+     * Removes a list of commands
+     * #### Examples:
+     * ````javascript
+     * var commands : annyang.CommandOption = {'hello': helloFunction, 'howdy': helloFunction, 'hi': helloFunction};
+     * // Add some commands
+     * annyang.addCommands(commands);
+     * // Don't respond to howdy or hi
+     * annyang.removeCommands(['howdy', 'hi']);
+     * ````
+     *
+     * @param {string[]} command
+     */
+    removeCommands(command: string[]): void;
```

Comment:
Does `removeCommands()` need to be repeated twice?

Could we do something like:
`removeCommands(command?: string | string[]): void;`

I am unsure of the syntax, so consider my suggestion pseudo code.